# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: gfx/layers/ImageDataSerializer.h
# Commit: 7a990fa02246
# Full Hash: 7a990fa022462c702adacfac907a0fc178940b9d
# Author: Jean-Yves Avenard <jyavenard@mozilla.com>
# Date: 2020-10-21 09:51:28
# Regressor Bug: 1595994
# File Overlap Count: 1
# Description:
#   Bug 1595994 - P1D. Properly serialize display size when sending image over IPC. r=mattwoodrow
#   
#   The code always assumed that the size of the image with the Y plane dimensions, which, while often the case, isn't correct.
#   We remove the assertions that the display offset was always (0,0) and properly carry the actual data over IPC.
#   
# ==============================================================================

diff -r 4c7f671c7b69 -r 7a990fa02246 gfx/layers/ImageDataSerializer.h
--- a/gfx/layers/ImageDataSerializer.h	Tue Oct 20 23:29:30 2020 +0000
+++ b/gfx/layers/ImageDataSerializer.h	Tue Oct 20 23:30:04 2020 +0000
@@ -11,6 +11,7 @@
 #include "mozilla/Attributes.h"             // for MOZ_STACK_CLASS
 #include "mozilla/RefPtr.h"                 // for already_AddRefed
 #include "mozilla/gfx/Point.h"              // for IntSize
+#include "mozilla/gfx/Rect.h"               // for IntRect
 #include "mozilla/gfx/Types.h"              // for SurfaceFormat
 #include "mozilla/layers/LayersSurfaces.h"  // for SurfaceDescriptor
 
@@ -57,6 +58,8 @@
 
 gfx::IntSize SizeFromBufferDescriptor(const BufferDescriptor& aDescriptor);
 
+gfx::IntRect RectFromBufferDescriptor(const BufferDescriptor& aDescriptor);
+
 Maybe<gfx::IntSize> CbCrSizeFromBufferDescriptor(
     const BufferDescriptor& aDescriptor);
 